### ISSUE TITLE
Responsive navigation: menu does not appear when hamburger clicked on

### ISSUE THREAD

**Issue by bobbingwide** [WordPress/gutenberg#33273]
## Description
In Gutenberg 11.0.0 the ability to view the navigation menu when it's reduced to a hamburger menu has disappeared.

It worked in 10.9.1.

It appears that the front end script is not being loaded in 11.0.0
```
<script type='text/javascript' src='https://s.b/wp55/tt1/wp-content/plugins/gutenberg/build/block-library/blocks/navigation/frontend.js?ver=5.7.2' id='core_block_navigation_load_frontend_scripts-js'></script>
```

## Step-by-step reproduction instructions

1. Using TT1 blocks Edit site and change the Navigation menu to responsive
OR add a new Horizontal navigation menu and set Enable responsive menu toggle to on. 
2. View the site and reduce the width until any responsive menu switches to a hamburger menu.
3. Try clicking on the hamburger to display the drop down menu.

## Expected behaviour
Menu appears when hamburger clicks.

## Actual behaviour
Menu doesn't appear.

## Screenshots or screen recording (optional)
Display during step 2.
![image](https://user-images.githubusercontent.com/2474435/124871118-9238b380-dfbb-11eb-87ff-36943fa06cf6.png)

Hamburgers appears.
With 11.0.0 you can't get any further.
![image](https://user-images.githubusercontent.com/2474435/124871255-b98f8080-dfbb-11eb-97ea-0d7715520bf7.png)

Expected result - with 10.9.1
![image](https://user-images.githubusercontent.com/2474435/124871414-ecd20f80-dfbb-11eb-88dd-07c0aa72c0eb.png)

## WordPress information
- WordPress version: 5.7.2
- Gutenberg version: 11.0.0
- Are all plugins except Gutenberg deactivated? Yes
- Are you using a default theme (e.g. Twenty Twenty-One)? TT1 blocks 

## Device information
- Device: Desktop
- Operating system: Windows
- Browser: Chrome Version 91.0.4472.124 (Official Build) (64-bit)

**Comment by bobbingwide** [comment]
The problem occurs in my Windows development machine. 
It doesn't occur on my Linux hosted servers where the JavaScript seems to be different.

```
<script type='text/javascript' src='https://cwiccer.com/wp-content/plugins/gutenberg/build/block-library/blocks/navigation/view.min.js?ver=5.7.2' id='wp-block-navigation-view-js'></script>
```

This is defined in `block.json`

```
"viewScript": "file:./view.min.js",
```

**Comment by bobbingwide** [comment]
It's a problem with Windows directory separators in `gutenberg_block_type_metadata_view_script()`
For the core/navigation block $metadata['file'] is   
`C:\apache\htdocs\wp55\wp-content\plugins\gutenberg\build\block-library\blocks/navigation/block.json"`

` gutenberg_dir_path()` returns `C:\apache\htdocs\wp55\wp-content\plugins\gutenberg/`

`strpos( $metadata['file'], gutenberg_dir_path() )` is false, not zero, so the script isn't enqueued. 

Disabling the test, the enqueued script is
```
<script type='text/javascript' 
src='https://s.b/wp55/thisis/wp-content/plugins/gutenberg/C:/apache/htdocs/wp55/wp-content/plugins/gutenberg/build/block-library/blocks/navigation/view.min.js?ver=5.7.2' id='wp-block-navigation-view-js'></script>
```







https://s.b/wp55/written/wp-content/plugins/gutenberg/C:/apache/htdocs/wp55/wp-content/plugins/gutenberg/build/block-library/blocks/navigation/view.min.js?ver=5.7.2

**Comment by bobbingwide** [comment]
### Possible fix

Change `gutenberg_dir_path()` to `call wp_normalize_path()`.
For this scenario it would return `C:/apache/htdocs/wp55/wp-content/plugins/gutenberg/`

Also wrap `$metdata['file']` and `$view_script_path` with `wp_normalize_path()`
for the `strpos()` and `str_replace()` calls involving `gutenberg_dir_path()`.

```
C:\apache\htdocs\wp55\wp-content\plugins\gutenberg>git diff -w lib/compat/wordpress-5.8/blocks.php
diff --git a/lib/compat/wordpress-5.8/blocks.php b/lib/compat/wordpress-5.8/blocks.php
index 54a9864a..528d5d12 100644
--- a/lib/compat/wordpress-5.8/blocks.php
+++ b/lib/compat/wordpress-5.8/blocks.php
@@ -23,12 +23,13 @@ function gutenberg_block_type_metadata_view_script( $settings, $metadata ) {
                ! isset( $metadata['viewScript'] ) ||
                ! empty( $settings['view_script'] ) ||
                ! isset( $metadata['file'] ) ||
-               strpos( $metadata['file'], gutenberg_dir_path() ) !== 0
+               strpos( wp_normalize_path( $metadata['file'] ), gutenberg_dir_path() ) !== 0
        ) {
            return $settings;
        }

        $view_script_path = realpath( dirname( $metadata['file'] ) . '/' . remove_block_asset_path_prefix( $metadata['viewScript'] ) );
+
        if ( file_exists( $view_script_path ) ) {
                $view_script_handle = str_replace( 'core/', 'wp-block-', $metadata['name'] ) . '-view';
                wp_deregister_script( $view_script_handle );
@@ -41,7 +42,7 @@ function gutenberg_block_type_metadata_view_script( $settings, $metadata ) {

                $result = wp_register_script(
                        $view_script_handle,
-                       gutenberg_url( str_replace( gutenberg_dir_path(), '', $view_script_path ) ),
+                       gutenberg_url( str_replace( gutenberg_dir_path(), '', wp_normalize_path( $view_script_path)  ) ),
                        $view_script_dependencies,
                        $view_script_version,
                        true
```

This should resolve the problem with the hamburger menu not appearing and
also problems with the file block in a Windows environment.

**Comment by annezazu** [comment]
cc @jasmussen as a heads up in case you can replicate this/have more context!

**Comment by jasmussen** [comment]
Thank you for the ping Anne. @vcanales does this ring any bells?

**Comment by vcanales** [comment]
This happens because TT1-blocks loads scripts before the body in the DOM, which means that when the responsive menu initialization was ran, it couldn't find any menus. [Fixed here.](https://github.com/WordPress/gutenberg/compare/fix/tt1-blocks-responsive-menu?expand=1)

This does bring up an interesting question about lazy loading; it would be necessary to load any lazily loaded menus in this manner as well. Something to keep in mind for the future.

**Comment by vcanales** [comment]
Something else to note is that the error may have been caused by failures to load the file in earlier versions, but [that has been handled.](https://github.com/WordPress/gutenberg/pull/32814)

**Comment by chthonic-ds** [comment]
I can confirm I'm seeing the same issue in a Windows environment. I didn't realise there was supposed to be a front-end script for the navigation toggle, so my theme has been supplying its own solution. Testing now in a Linux environment, I see `view.min.js` is present.

WordPress 5.8.1
Gutenberg 11.4.1

**Comment by bobbingwide** [comment]
I've got the same symptoms again with WordPress 6.1 and Gutenberg 14.4.0 & 14.5.0.
Clicking on the hamburger menu does nothing in a Windows local development environment.
Clicking on the hamburger menu works for sites hosted on Linux. 
Currently investigating as https://github.com/bobbingwide/written/issues/19

**Comment by paaljoachim** [comment]
Reopening isse as @bobbingwide Herb reported that the problem has showed up again in Gutenberg 14.5.0 and WordPress 6.1 in a Windows environment.

**Comment by bobbingwide** [comment]
Problem reproduced today with TwentyTwenty-Three.
Neither `/navigation/view.min.js` nor `navigation/view-modal.min.js` are being enqueued.

**Comment by t-hamano** [comment]
I also tested it in a local environment on a Windows OS.
In my case, when I activate the Gutenberg plugin, only `view-modal.min.js` is not enqueued.

- OS: Windows 11
- WordPress Version: 6.1
- Gutenberg Plugin Version: 14.5.0
- Theme: Twenty Twenty Three

<details>
<summary>Without Gutenberg Plugin</summary>

![wthout](https://user-images.githubusercontent.com/54422211/201913675-d8844719-e311-4d98-a14a-b7503a1892b0.png)

</details>

<details>
<summary>With Gutenberg Plugin</summary>

![with](https://user-images.githubusercontent.com/54422211/201913694-8fe6493d-4001-4158-b6eb-3a7abf102e22.png)

</details>

**Comment by t-hamano** [comment]
I have identified the cause of the problem.
As you can see from[ this comment](https://github.com/WordPress/gutenberg/issues/33273#issuecomment-877304927), in places the path separator characters did not match.
I will submit a PR in the near future.

**Comment by bobbingwide** [comment]
@t-hamano thanks for the confirmation. 

I've proposed a  new fix in https://github.com/bobbingwide/written/issues/19#issuecomment-1315267873
It's not quite the same problem as in my original quick fix.  
I found I needed to use `wp_normalize_path()` more than before.
Would you care to try it? I don't have the capability to create a PR right now.

**Comment by t-hamano** [comment]
@bobbingwide 
Thanks for the detailed analysis!
I am about to write a description of #45773 and am making almost the exact same changes you suggested.

**Comment by vcanales** [comment]
Thanks for looking into this! I'll be around to review :)

**Comment by t-hamano** [comment]
@bobbingwide @vcanales 

The PR #45773 is ready. If you can test this PR in a Windows environment, I would be glad to have you review it.

**Comment by draganescu** [comment]
Closed by #45773
